server/service/src/static_files.rs

```diff
@@ -70,11 +70,14 @@ impl StaticFileService {
     // at the time of method creation TempFile only comes from web multipart
     pub fn move_temp_file(
         &self,
-        temp_file: TempFile,
+        temp_file: &TempFile,
         category: &StaticFileCategory,
         file_id: Option<String>,
     ) -> anyhow::Result<StaticFile> {
-        let file_name = temp_file.file_name.context("Filename not provided")?;
+        let file_name = temp_file
+            .file_name
+            .clone()
```

Comment:
```suggestion
            .to_owned()
```
It doesn't matter at all, but this might be slightly more correct, as you're using clone here to turn a reference into an `owned` value. The code underneath will probably be exactly the same but thought I'd mention anyway.